server/src/main/java/org/elasticsearch/index/mapper/LuceneDocument.java

```diff
@@ -74,9 +74,14 @@ public void addAll(List<? extends IndexableField> fields) {
     }
 
     public void add(IndexableField field) {
+        assert assertLegalFieldName(field);
+        fields.add(field);
+    }
+
+    private boolean assertLegalFieldName(IndexableField field) {
```

Comment:
static?

Reply:
This one uses this `this.prefix` unfortunately.